src/modules/hip/kernel/non_silent_region_detection.hpp

```diff
@@ -0,0 +1,406 @@
+                                                      Rpp32s *detectionLengthTensor,
+                                                      Rpp32f cutOffDB,
+                                                      Rpp32s windowLength,
+                                                      Rpp32f referencePower,
+                                                      Rpp32s resetInterval,
+                                                      rpp::Handle& handle)
+{
+    // allocate temporary memory for MMS Array
+    Rpp32f *mmsArr;
+    hipMalloc(&(mmsArr), srcDescPtr->n * srcDescPtr->strides.nStride * sizeof(Rpp32f));
+
+    int maxSharedMemoryInBytes = 32000; // 32 KB
+    int maxSharedMemoryElements = maxSharedMemoryInBytes / sizeof(Rpp32f);
+    int kSharedMemBanks = 32;
+    int inputTileLength = prev_pow2(maxSharedMemoryElements * kSharedMemBanks / (kSharedMemBanks + 1));
+
+    if (resetInterval > 0 && resetInterval < inputTileLength)
+    {
+        auto p = prev_pow2(resetInterval);
+        auto n = next_pow2(resetInterval);
+        if (p > windowLength)
+            inputTileLength = p;
+        else if (n < inputTileLength)
+            inputTileLength = n;
+    }
+
+    int sharedMemorySizeInBytes = smem_pos(inputTileLength) * sizeof(Rpp32f);
+    int outputTileLength = inputTileLength - windowLength;
+    float windowFactor = 1.0f / windowLength;
+
+    if (outputTileLength <= 0)
+    {
+        std::cout << "Invalid output tile length! " << std::endl;
+        return RPP_ERROR;
+    }
+    if (sharedMemorySizeInBytes > maxSharedMemoryInBytes)
+    {
+        std::cout << "Cannot compute the requested moving mean square, due to shared memory restrictions" << std::endl;
+        return RPP_ERROR;
+    }
+
+    // launch kernel to compute the values needed for MMS Array
+    int globalThreads_x = std::min<int>(ceil(static_cast<float>(srcDescPtr->strides.nStride) / outputTileLength), 1024);
+    int globalThreads_y = 1;
+    int globalThreads_z = handle.GetBatchSize();
+    int localThreads_x = 512;
+    int localThreads_y = 1;
+    int localThreads_z = 1;
+
+    hipLaunchKernelGGL(moving_mean_square_hip_tensor,
+                       dim3(globalThreads_x, globalThreads_y, globalThreads_z),
+                       dim3(localThreads_x, localThreads_y, localThreads_z),
+                       sharedMemorySizeInBytes,
+                       handle.GetStream(),
+                       srcPtr,
+                       srcDescPtr->strides.nStride,
+                       mmsArr,
+                       srcLengthTensor,
+                       outputTileLength,
+                       windowLength,
+                       windowFactor,
+                       inputTileLength);
+    hipStreamSynchronize(handle.GetStream());
+
+    const Rpp32f cutOff = std::pow(10.0f, cutOffDB * 0.1f);
+    bool referenceMax = (referencePower == 0.0f);
+    Rpp32f *partialMaxArr = handle.GetInitHandle()->mem.mgpu.maskArr.floatmem;
```

Comment:
Rename to scratchBufferHip. Pls merge develop into your branch since there's no maskArr anymore